src/gurt/telemetry.c

```diff
@@ -704,6 +710,42 @@ create_shmem(const char *root_path, key_t key, size_t size_bytes,
 	return 0;
 }
 
+int
+destroy_shmem_with_key(key_t key)
+{
+	struct d_tm_shmem_hdr		*header;
+	struct shmem_region_list	*entry;
+	d_list_t			*cur;
+	d_list_t			*head;
+	int				rc;
+	int				shmid;
+
+	rc = open_shmem(key, &header);
+	if (rc == -DER_NO_SHMEM) /* if it doesn't exist, nothing to do */
+		return 0;
+	if (rc < 0) {
+		D_ERROR("Unable to open shmem region 0x%x for cleanup. An admin must clean up "
+			"manually using ipcrm.\n", key);
```

Comment:
This is good. We might even want to consider adding a new RAS event at some point so that this situation is clearly identified in syslog, too.